src/Servers/Kestrel/test/FunctionalTests/UnixDomainSocketsTests.cs

```diff
@@ -63,6 +65,10 @@ async Task EchoServer(ConnectionContext connection)
                     {
                         Logger.LogDebug("Graceful shutdown triggered for {connectionId}.", connection.ConnectionId);
                     }
+                    finally
+                    {
+                        serverConnectionCompletedTcs.TrySetResult(null);
```

Comment:
Unless the connection closes before this finally, I don't think this will fix the race.

Reply:
There's no race.

Reply:
Talked offline, it's due to the 5-second timeout, not because this is racey.